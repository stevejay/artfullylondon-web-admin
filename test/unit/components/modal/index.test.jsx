import React from 'react'
import { shallow } from 'enzyme'
import _ from 'lodash'

import Modal from '_src/components/modal'

const TestTransition = props => <div id='transition' />

it('should render correctly', () => {
  const wrapper = shallow(
    <Modal show transition={TestTransition} onHide={_.noop}>
      <div id='child' />
    </Modal>
  )

  expect(wrapper).toMatchSnapshot()
})

it('should ignore update when props have not changed', () => {
  const wrapper = shallow(
    <Modal show transition={TestTransition} onHide={_.noop}>
      <div id='child' />
    </Modal>
  )

  const result = wrapper.instance().shouldComponentUpdate({ show: true })
  expect(result).toEqual(false)
})

it('should trigger update when props have changed', () => {
  const wrapper = shallow(
    <Modal show transition={TestTransition} onHide={_.noop}>
      <div id='child' />
    </Modal>
  )

  const result = wrapper.instance().shouldComponentUpdate({ show: false })
  expect(result).toEqual(true)
})
