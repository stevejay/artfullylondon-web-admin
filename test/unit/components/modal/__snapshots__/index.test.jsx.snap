// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<Modal
  aria-label="Quicksearch"
  autoFocus={true}
  backdrop={true}
  backdropClassName="backdrop"
  backdropTransition={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={true}
  transition={[Function]}
>
  <div
    id="child"
  />
</Modal>
`;
